packages/server/modules/core/domain/objects/types.ts

```diff
@@ -11,7 +11,8 @@ export type SpeckleObject = ObjectRecord
  */
 export type InsertableSpeckleObject = NullableKeysToOptional<
   OverrideProperties<
-    SetOptional<SpeckleObject, 'createdAt'>,
+    // Both createdAt & speckleType have defaults upon insertion
```

Comment:
@didimitrie @gjedlicska Ended up making these changes, I think pretty much all of RawSpeckleObject is optional

Reply:
see comment below, I understand, this is inline with the DB schema, but the default for `speckleType` has bit us in the past.
There was a bug in the code, where we were specifying a value for `speckle_type` but not for `speckleType` and everything defaulted to `Base`
 TBF i'd like to remove the default and make it mandatory.